babel.config.js

```diff
@@ -1,29 +1,16 @@
-module.exports = {
-  env: {
-    test: {
-      presets: [
-        [ '@babel/env', { loose: true } ],
-        '@babel/react'
-      ],
-      plugins: [
-        '@babel/transform-runtime'
-      ]
-    },
-    cjs: {
-      presets: [
-        [ '@babel/env', { loose: true } ]
-      ],
-      plugins: [
-        '@babel/transform-runtime'
-      ]
-    },
-    esm: {
-      presets: [
-        [ '@babel/env', { loose: true, modules: false } ]
-      ],
-      plugins: [
-        ['@babel/transform-runtime', { useESModules: true }]
-      ]
-    },
+module.exports = function getConfig(api) {
+  const isRollup = api.caller(
+    caller => caller && caller.name === 'rollup-plugin-babel'
+  )
+  if (!isRollup)
+    return {
+      presets: [['@babel/preset-env', { loose: true }], '@babel/react'],
+    }
+  return {
+    presets: [['@babel/preset-env', { loose: true, modules: false }]],
+    plugins: [
+      'babel-plugin-annotate-pure-calls',
+      ['@babel/transform-runtime', { useESModules: true }],
```

Comment:
I thing is missing here. cjs bundle will get esm imports.

Here I added transform-runtime in place and didin't use it for tests and benchmarks. babel runtime is just an optimisation.
https://github.com/istarkov/rifm/blob/master/rollup.config.js#L19